src/reanimated2/layoutReanimation/web/animationsManager.ts

```diff
@@ -96,14 +110,16 @@ function tryGetAnimationConfigWithTransform<
     typeof config.constructor;
 
   const isLayoutTransition = animationType === LayoutAnimationType.LAYOUT;
+  const isCustomKeyframe = config instanceof Keyframe;
   const initialAnimationName =
     typeof config === 'function'
       ? config.presetName
       : (config.constructor as ConstructorWithStaticContext).presetName;
 
   const shouldFail = checkUndefinedAnimationFail(
     initialAnimationName,
-    isLayoutTransition
+    isLayoutTransition,
+    isCustomKeyframe
```

Comment:
We could merge those (`isLayoutTransition || isCustomKeyframe`) into one flag. Let me know what you think!